docs/en/reference/working-with-associations.rst

```diff
@@ -511,9 +511,23 @@ Thanks to ``cascade: remove``, you can easily delete a user and all linked comme
     overhead, especially when the cascaded collections are large. Make sure
     to weigh the benefits and downsides of each cascade operation that you define.
 
-    To rely on the database level cascade operations for the delete operation instead, you can
-    configure each join column with :doc:`the onDelete option <working-with-objects>`.
+    If you can go without lifecycle events, you can rely on database-level foreign keys
+    with cascade operations instead. To do so, configure each join column with 
+    :doc:`the onDelete option <working-with-objects>` and remove the ``cascade: remove`` setting.
+    
+    Contrary to in-memory cascade operations, those at the database level are also applied 
+    when using :doc:`DQL DELETE statements<dql-doctrine-query-language#delete-queries>`.
+
+.. warning::
 
+    For many-to-many associations, ``cascade: remove`` means that the associated entities
```

Comment:
The same is true for one-to-many though, are we sure this warning here is necessary? It repeats what the previous docs stated in other words. Maybe the words in the beginning of the section could need more precision: "Thanks to cascade: remove, you can easily ...."

Reply:
In a one-to-many association (e.g. person -> address) I find it natural that removing the person also removes their addresses.

In many-to-many, I was somehow surprised (although I should have known better) that „remove“ not only affects the n:m-table, but the entity at the other side of the association.

Would Doctrine even go back across the association? E. g. remove book -> remove author(s) -> remove other books by these authors?

Reply:
Maybe a code example of the Book <-> Author relation can help some readers to understand what's happening in the example with the configuration.

Reply:
Not sure how to best proceed here... suggestions anyone?

Reply:
I'd use an example that is more clearly a many to many. While books sometimes have several authors, that's rarely the case. How about an example with a relationship between developers and projects? I must admit I'm a bit surprised by this behavior as well, but in hindsight, it does make sense: if you don't cascade, you're still going to remove n-m table entries, so of course cascade does a bit more.

I think @beberlei's suggestion might make the first sentence a bit too big and distracting, I wouldn't move all your warning to there, but I would  maybe change the tone to a more startling one:

```rst
``cascade:remove`` is to ``EntityManager::remove()`` what ``-rf`` is to ``rm``, so please read carefully.
```